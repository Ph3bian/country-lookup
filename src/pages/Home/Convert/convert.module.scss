@import "../../../styles/generic";

.Convert {
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  height: 100vh;
  width: 20rem;
  background: white;
  box-shadow: 0 40px 64px 0 rgba(65, 78, 101, 0.1), 0 24px 32px 0 rgba(65, 78, 101, 0.1),
    0 16px 16px 0 rgba(65, 78, 101, 0.1), 0 8px 8px 0 rgba(65, 78, 101, 0.1), 0 4px 4px 0 rgba(65, 78, 101, 0.1),
    0 2px 2px 0 rgba(65, 78, 101, 0.1);
    z-index: 1;
  &Header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 3.5rem;
    background: $primary;
    color: $white;
    padding: 1rem;
    span {
      cursor: pointer;
      font-size: 0.875rem;
      svg{
        fill: $white
      }
    }
  }
  &Body {
    height: calc(100vh - 3.125rem);
    padding: 2rem;
    label{
      color:#757575;
    }
    button {
      background: $secondary;
      width:100%;
    }
  }
  
}
