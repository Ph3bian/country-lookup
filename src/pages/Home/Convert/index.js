import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Input, Button } from 'components/Form'
import { useToasts } from 'components/Toaster'
import { Close } from 'assets/svg'
import styles from './convert.module.scss'

const Convert = ({ setAmount, closeConvertSlider }) => {
    const { addToast } = useToasts()
    const [convertAmount, setConvertAmount] = useState(0)
    let error = ''
    const handleSubmit = e => {
        e.preventDefault()
        if (convertAmount <= 0) {
            return addToast('Enter valid amount', { appearance: 'info' })
        }
        if (convertAmount.match(/^\d+\.?\d*$/) === null) {
            error = 'Enter valid amount'
            return addToast('Invalid amount', { appearance: 'info' })
        }
        setAmount(convertAmount)
        return closeConvertSlider(false)
    }

    const closeConvert = () => closeConvertSlider(false)

    return (
        <div className={styles.Convert}>
            <div className={styles.ConvertHeader}>
                <p>Convert Amount</p>
                <span onClick={closeConvert}>
                    <Close />
                </span>
            </div>
            <form onSubmit={handleSubmit}>
                <div className={styles.ConvertBody}>
                    <Input
                        name={'amount'}
                        value={convertAmount}
                        onChange={e => setConvertAmount(e.target.value)}
                        label="Amount"
                        type={'text'}
                        error={error}
                    />
                    <Button type={'submit'} value={'Submit'} />
                </div>
            </form>
        </div>
    )
}
Convert.propTypes = {
    setAmount: PropTypes.func,
    closeConvertSlider: PropTypes.func
}
export default Convert
