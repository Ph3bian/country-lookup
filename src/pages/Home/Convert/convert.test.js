import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { ToastProvider } from 'react-toast-notifications'
import Convert from './index'

let container = null
beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
})
afterEach(() => {
    unmountComponentAtNode(container)
    container.remove()
    container = null
})

it('renders with or without props', () => {
    act(() => {
        render(
            <ToastProvider>
                <Convert />
            </ToastProvider>,
            container
        )
    })
    const header = container.querySelector('p')
    expect(container.querySelector('form')).toBeDefined()
    expect(container.querySelector('button').innerHTML).toBe('Submit')
    expect(header.textContent).toBe('Convert Amount')
})


