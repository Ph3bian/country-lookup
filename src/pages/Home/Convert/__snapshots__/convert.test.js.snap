// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props function closeConvertSlider 1`] = `undefined`;
